src/factory.ts

```diff
@@ -410,8 +410,11 @@ function sanityCheck(factory: FactoryGraph): void {
  */
 export function buildFactory(
     requirements: Map<Craftable, { count: number; maintain: number }>,
+    factory: FactoryGraph | undefined,
```

Comment:
Optional parameters can be declared using `?`
```suggestion
    factory?: FactoryGraph,
```